src/factories/calorimetry/CalorimeterClusterRecoCoG_factoryT.h

```diff
@@ -22,13 +22,24 @@ class CalorimeterClusterRecoCoG_factoryT :
         std::string tag,
         const std::vector<std::string>& input_tags,
         const std::vector<std::string>& output_tags,
-        CalorimeterClusterRecoCoGConfig cfg)
-    : JChainMultifactoryT<CalorimeterClusterRecoCoGConfig>(tag, input_tags, output_tags, cfg),
+        ConfigMap cfg)
+    : JChainMultifactoryT<ConfigMap>(tag, input_tags, output_tags, cfg),
       m_algo(tag) {
 
       DeclarePodioOutput<edm4eic::Cluster>(GetOutputTags()[0]);
       DeclarePodioOutput<edm4eic::MCRecoClusterParticleAssociation>(GetOutputTags()[1]);
 
+      // Initialize properties
+      for (const auto& [key, prop] : m_algo.getProperties()) {
+        if (cfg.count(std::string(key)) > 0) {
```

Comment:
I feel like getting an iterator once would be better than triple lookup we do now.

Reply:
c4c6d86dd15fb4cff91a8d0faa617e1d9e490cf2